### PR TITLE
Remove Dependencies.swift

### PR DESCRIPTION
**Pull request by fortmarek**
Resolves https://github.com/tuist/tuist/issues/5712

### Short description 📝

We're removing the `Dependencies.swift` manifest in favor of `Package.swift`.

I also took this opportunity to remove the intermediary `graph.json` we were creating and instead we now load it on demand. To ensure `tuist generate` speed doesn't degrade, we cache `Package.swift` the same way we cache any other manifest. So, first `tuist generate` can take a bit longer but then on hot runs, it should be really fast as we read only the dumped JSON representations. Users would take this initial hit before as well, it would only be earlier in the `tuist fetch`.

### How to test the changes locally 🧐

Run `tuist fetch` and `tuist generate` on a project with `Package.swift` manifest only (e.g. the `app_with_spm_dependencies` fixture - we also test this one, among others, in our acceptance tests)

### Contributor checklist ✅

- [x] The code has been linted using run `mise run lint:fix`
- [x] The change is tested via unit testing or acceptance testing, or both
- [x] The title of the PR is formulated in a way that is usable as a changelog entry
- [x] In case the PR introduces changes that affect users, the documentation has been updated

### Reviewer checklist ✅

- [x] The code architecture and patterns are consistent with the rest of the codebase
- [x] Reviewer has checked that, if needed, the PR includes the label `changelog:added`, `changelog:fixed`, or `changelog:changed`, and the title is usable as a changelog entry

### CODE REVIEW

File: Sources/TuistKit/Services/DumpService.swift [hunk 1]
```diff
@@ -37,8 +37,6 @@ final class DumpService {
             encoded = try manifestLoader.loadConfig(at: projectPath.appending(component: Constants.tuistDirectoryName))
         case .template:
             encoded = try manifestLoader.loadTemplate(at: projectPath)
-        case .dependencies:
-            encoded = try manifestLoader.loadDependencies(at: projectPath)
```
Comment (danyf90) [inline Sources/TuistKit/Services/DumpService.swift:41]: Shouldn't we dump the config we have in `PackageSettings` config we have in `Package.swift`

**Review by danyf90 — APPROVED** [review]

File: Sources/TuistKit/Services/DumpService.swift (hunk 1 above)
Comment (fortmarek) [reply Sources/TuistKit/Services/DumpService.swift:41]: Could you rephrase? I'm not sure I follow

Comment (danyf90) [reply Sources/TuistKit/Services/DumpService.swift:41]: In tuist 3 I can do `tuist dump dependencies` and get a bump of the dependencies and their configuration. How would i do it in tuist 4?

Comment (fortmarek) [reply Sources/TuistKit/Services/DumpService.swift:41]: I will add a command to dump the `PackageSettings` model 👍

Comment (fortmarek) [reply Sources/TuistKit/Services/DumpService.swift:41]: done 🙌
